### ISSUE TITLE
Add support to -fprofile-prefix-path=path

### ISSUE THREAD

**Issue by jdavidpeter** [ccache/ccache#1507]
TLDR:
Current CCache (4.10.2) doesn't support shared caching for coverage builds in different build workspaces using gcc.
By adding support for -fprofile-prefix-path=path gcc option and making sure that "apparent real path" of the object file is not hashed when profile-prefix-path is set, shared caching for coverage could be enabled.

Useful gcc links:
https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html
https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html

Context:

In CI environments ccache can drastically increase build space for PR validation builds.
As many CI environments operate on bare metal and are placing workspaces to different directories, caching is only possible if references to build absolute paths in object files are removed.
Using gcc on Linux systems, this is trivial for release builds, easy to achieve for debug (using -ffile-prefix-map) but is seemingly impossible for coverage builds, which produce gcno files build time and gcda files during execution.

For coverage, the compiler generates a gcno file on top of the object file, which is cached and supported by ccache. However, gcc also bakes in the runtime gcda file path in the object file, where the runtime coverage information will be recorded.
By default, this is an absolute path pointing to the same directory as the result object file, so cross-workspace caching is not possible.

There is a build time option, -fprofile-dir=path that can instruct gcc to bake in a different gcda file path in the object files compile time. This new path can be relative, therefore independent of build workspace paths. This option is also supported by ccache.
However, when using this option with realitve paths, the resulting gcda file names will contain a mangled absolute path to the build workspace to avoid potential filename collisions when all gcda files are put in the same folder during execution.
This again, prevents reuse of cache across build workspaces for coverage and I believe was the main reason for a bugfix in version 4.10 which started to hash the "apparent real path" of the object file in the cache key.

However, gcc offers yet another option, -fprofile-prefix-path=path.
When -fprofile-dir is set to a relative path, -fprofile-prefix-path can be set to the root of the build workspace and the absolute path of the build workspace will be cut from the gcda file name baked in the object file.
This makes the resulting gcda file name independent of the build workspace absolute path and thus would enable cache reuse of coverage builds across different build workspaces, e.g. in CI environments.

Unfortunately, -fprofile-prefix-path is not supported by ccache and it just bails when detecting it.

Would it be possible to add support for it?
If so, what parts of the code would need change? I could create a PR if maintainers think this fits to the vision and give me some pointers to start with.

**Comment by jdavidpeter** [comment]
I'm on the way to complete this request, will raise a PR soon.
